src/librustc/hir/mod.rs

```diff
@@ -159,7 +159,8 @@ impl fmt::Debug for Lifetime {
 
 impl Lifetime {
     pub fn is_elided(&self) -> bool {
-        self.name == keywords::Invalid.name()
+        self.name == keywords::Invalid.name() ||
+        self.name == "'_"
```

Comment:
So I've never been crazy about using a "sentinel value" here rather than an enum -- now that we've got more than one sentinel, I feel even less good. Can't we change this to something like

```rust
enum LifetimeName {
    Elided,
    Underscore,
    Static,
    Name(Token) // or whatever this is
}
```

Reply:
It's a `Symbol` / `ast::Name` I think.